group network api 리팩토링

theBettor opened the pull request:
## 🤷‍♂️ PR 내용(Issue 꼭 달기!)
resolved: #85 
'사용자의 그룹 목록 조회', '그룹의 초대코드 조회' 추가 작업을 진행했습니다.
issue 내용이 좀 어지럽습니다. 요약하면

> DataSource에 새로 추가한 2개의 @get suspend 함수들이 unused하고 있으니 어딘가에서 호출해줘야겠다. 보니까 core/data/repository랑 test쪽에서 호출하고 있음.

  -> 그런데 repository를 추가하면서 domain usecase까지 만들어야하는 것 같아서 맞게 하고 있는지 이런식으로 진행하면 되는지 다양하게 코멘트를 받는게 좋을듯하여 pr만듭니다. 감사합니다!

1198f19bc1c1b60c2a8bb0c24b86f6a9a043ba22
여기서부터 보시면 됩니다..
![스크린샷 2025-01-11 오전 12 56 42](https://github.com/user-attachments/assets/8a4b90f4-647e-496a-9456-59cdb0fd40c7)

## -> UnitTest 관련 추가 작업은 일단은 하지 않는것으로 결정

theBettor committed:
test: 테스트

theBettor committed:
OnboardingNavigation.kt 에러를 잡고 싶습니다.. Home쪽 참고하면서 하고 있는데 Onboarding 모듈 내에서 해결해야하는 문제는 아닌거같아요.. #80

theBettor committed:
refactor: build-logic의 feature 플러그인 추가

theBettor committed:
refactor: 어떻게 하면 OnboardingPermissionNotificationScreen로 화면이동 할 수 있을까 고민인데.. ViewModel에 문제가 생긴듯 하다

theBettor committed:
refactor: 로그인이 필요한 경우의 메서드 수정

기존 메서드에 navController가 안들어가는 문제로 메서드를 삭제하고 initView() 메서드에 해당 flow를 접목하여 LaunchedEffect로 온보딩화면으로 이동하게 바꿈

theBettor committed:
refacotr: 해당 컴포넌트 필요없어서 삭제

theBettor committed:
refactor: 온보딩 navigation 수정

figma가 완성은 되지 않았지만, 흐름은 대강 이렇다.

-로그인 되지 않은 경우
FirstScreen 보여주기
- 로그인o, 권한x
PermissionNotificationScreen으로 이동
- 로그인o, 권한o
StartScreen

남은 문제
1. 뷰모델에서 알림권한 업데이트 로직을 수정(+ 하모니 시작하기 버튼)
2. 로그인o, 권한o 말고도 그룹까지 속해져있으면 당연히 Home으로 가야하는데 관련 로직이 없음
-> 그룹에 속해 있지 않을 경우, StartScreen을 띄우게 해야함.
3. 그룹 입장하기 했을 때, EnterScreen으로 가는 onClick메서드 추가하기
4. EnterScreen 부분 마무리하기

설명:
OnboardingRoute가 기본이 되며, 실제로 다른 화면을 보여줘야하는 state일때는 알림권한화면과 그룹화면이여서 2개의 루트를 추가로 만들어 파라미터로 갖고 있음 각각 Navigation을 설정해주고 MainNavHost에 관련 코드를 추가해주어야함. 아직 개념이 잘 안잡히긴했는데 이런 흐름~

theBettor committed:
refactor: 미완성된 뷰모델 수정(+ 테스트 방법)

1.loggedInDebug를 사용하면 로그인이 다 되었다고 인지하기에 Home으로 감
2.지금은 webpage not available이 뜨는데, viewmodel의 runCatching의 body를 주석처리하고 테스트하면 됨.
3. FirstScreen의 LaunchedEffect 잘 유념하면서 수정해보자.

theBettor committed:
refactor: 그룹 만들기/입장 분리

계속 수정해야하나, 그룹 만들기는 기획이 마무리 안되어서 클래스만 만들어놓음

theBettor committed:
feat: 알림권한 뷰모델이 아닌 LaunchedEffect로 관리

1. 뷰모델을 사용하지 않고 LaunchedEffect 내에서 알림 권한을 필요로하는 권한 설정하는 변수를 선언하고, 로그인o / 권한x 일 때 화면이동하는 로직으로 구성.
2. 버튼을 눌렀을때 POST_NOTIFICATIONS를 설정하는 코드도 추가
3. Manifest에서 uses-permission도 추가

theBettor committed:
feat: 알림권한 뷰모델이 아닌 LaunchedEffect로 관리

theBettor committed:
remove: 오타 수정

theBettor committed:
Merge remote-tracking branch 'origin/refactor/onboarding' into refactor/onboarding

theBettor committed:
refactor: 누락된 로직 추가

로그인o / 권한o 일때 StartScreen으로 보내는 로직 추가

theBettor committed:
refactor: 뷰모델 수정

알람권한 관련 로직 삭제, LoginUseCase랑 errorMessage는 나중에 적용해야될 것 같아서 북마크 해두었음.

theBettor committed:
feat: 2개의 Route 연결

1.Navigation에서 StartScreen의 파라미터인 MakeGroup, Enter를 Route를 만든다.
2. Navigation StartRoute NavGraphBuilder에 2개의 경로를 파라미터로 추가해주고, 'Route'를 설정해준다,.
3. 이미 만든 2개의 경로의 NavGraphBuilder에는 이동할 '화면'을 지정해준다.

4. OnboardingStartRoute를 기반으로 하는 화면 이동이기에 Route Class에 @Composable internal fun Route를 생성해준다.
5. MainNavHost에 기존에 생성되었던 addStartScreen에 추가된 파라미터 2개를 추가하며 navigation 설정을 해준다. 바로 밑에 2개의 화면을 호출해준다(NavGraphBuilder).

theBettor committed:
refacotr: OnboardingLayout 수정

생각보다 개판인게 많아서 더 수정해야함.

theBettor committed:
refactor: EnterScreen 수정

우선으로는 OnboardingLayout 구조를 바꿔서 프로필?이미지 위치를 재조정이 필요함.

theBettor committed:
feat: 초대코드 입력 후 확인버튼 누르면 화면 이동

1. 상태, 버튼 클릭, 무엇이든 화면이동을 하고 싶으면 Internal fun Route() 만들어야함. -> '이동해야하는 곳의 Route'를 정의한다.
2. 근간이 되던 OnboardingEnterRoute를 수정해야하는데, '어떤 화면에서 어떤 동작(parameter)을 시작할건지'를 정의하고, '이동해야 하는 곳의 Route'를 data object로 생성하고 '기본이 되는 Route'에서 '이동해야하는 화면'으로 정의해준다.
3. MainNavHost에서도 수정해줘야 하는데, 동일하게 '어떤 동작을 할건지 수정'해주고, '이동해야하는 화면'을 호출한다.

theBettor committed:
작업중

theBettor committed:
refactor : GroupInviteResponse 네이밍 변경

theBettor committed:
feat: '사용자의 그룹 목록 조회', '그룹의 초대코드 조회' 추가

theBettor committed:
feat: repository 추가해주려니까 domain usecase까지 만들라고함.. 이렇게 작업해도 되는건지해서 중단하고 pr올려서 코멘트 받고자함.

theBettor committed:
문제 있는 부분 잠시 주석

theBettor committed:
문제 있는 부분 잠시 주석

theBettor committed:
코드 컨벤션 수정

DJDrama commented on app/src/main/java/com/teampatch/harmony/MainActivity.kt:
```diff
@@ -66,9 +65,9 @@ class MainActivity : ComponentActivity() {
     private fun initView() = setContent {
         HarmonyTheme {
-            val navController: NavHostController = rememberNavController()
+            val navController: NavHostController = rememberNavController1()
```
rememberNavController1 

왜 1을 붙이신거에요??

theBettor replied:
저게 아마 실수로 입력된 것 같은데 다른 브랜치에서 작업하던 내용이라 참고해서 수정하겠습니다!

theBettor committed:
refactor: ktlintFormat으로 수정

theBettor committed:
Merge remote-tracking branch 'origin/refactor/group_network_api' into refactor/group_network_api

theBettor committed:
refactor: ktlintFormat으로 수정

theBettor committed:
refactor: repository interce 추가 및 수정

두 가지 모두 일단은 String으로 반환하는게 맞는 것 같고, queryUserGroupList의 경우는 interface없이 구현체만 있어 override를 지워주었음.

theBettor committed:
refactor: rememeberNavController1은 작업 중 실수로 오타가 생긴듯 하다.

theBettor committed:
Merge branch 'develop' into refactor/group_network_api

theBettor committed:
refactor: conflict 해결

agvber commented on core/domain/src/main/java/com/teampatch/core/domain/repository/GroupManagementRepository.kt:
```diff
@@ -10,10 +10,15 @@ interface GroupManagementRepository {
 
     /**
-     * @return Group 초대 코드
+     * @return Group 초대 코드 재생성
```
리턴 항목이 초대 코드 재생성이 좋을까요??

제가 생각하기에는 초대 코드가 더 좋을 것 같아요!

theBettor replied:
저게 스웨거에서도 regenerate 라는 키워드를 사용하고 있어서 어느 장단에 맞춰야하는지 고민이었긴 합니다..

agvber commented on core/domain/src/main/java/com/teampatch/core/domain/repository/GroupManagementRepository.kt:
```diff
@@ -10,10 +10,15 @@ interface GroupManagementRepository {
     suspend fun joinFamilyGroup(inviteCode: String): InvitedGroup
+
+    /**
+     * @return Group 초대 코드 조회
```
리턴 항목이 초대 코드 조회인데 
정확히 이해되지 않습니다

theBettor committed:
refactor: Repository interface 함수들의 정의와 리턴값을 명확히 제시하기
